### PR TITLE
Add the renderable interface.

### PR DESCRIPTION
**Pull request by Camerooooon**

### CODE REVIEW

File: src/org/dhsdev/flowerknight/FlowerKnight.java [hunk 1]
```diff
@@ -50,7 +54,9 @@ public static void init() {
 
         Camera.init();
 
-        TestImage.init();
+        //TestImage.init();
```
Comment (Sophon96) [inline src/org/dhsdev/flowerknight/FlowerKnight.java:57]: Delete?

**Review by adamhutchings — APPROVED** [review]

File: src/org/dhsdev/flowerknight/FlowerKnight.java (hunk 1 above)
Comment (JakeRoggenbuck) [reply src/org/dhsdev/flowerknight/FlowerKnight.java:57]: Yea

File: src/org/dhsdev/flowerknight/FlowerKnight.java [hunk 2]
```diff
@@ -88,14 +100,16 @@ public static void mainloop() {
      */
     public static void exit() {
 
-        TestImage.delete();
+        for (Renderable renderable : Renderable.renderables) {
+            renderable.clear();
+        }
+        //TestImage.delete();
```
Comment (JakeRoggenbuck) [inline src/org/dhsdev/flowerknight/FlowerKnight.java:106]: Same here

File: src/org/dhsdev/flowerknight/FlowerKnight.java [hunk 3]
```diff
@@ -72,7 +78,13 @@ public static void mainloop() {
             window.clear();
 
             // Render test
-            TestImage.render();
+            //TestImage.render();
```
Comment (Sophon96) [inline src/org/dhsdev/flowerknight/FlowerKnight.java:81]: And here

**Review by adamhutchings — APPROVED** [review]
Hi Cameron

**Review by JakeRoggenbuck — APPROVED** [review]
Works on linux!

File: src/org/dhsdev/flowerknight/gl/Renderable.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package org.dhsdev.flowerknight.gl;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a renderable object
+ * @author cameron
+ */
+public interface Renderable {
+    public static ArrayList<Renderable> renderables = new ArrayList<Renderable>();
```
Comment (Shuzhengz) [inline src/org/dhsdev/flowerknight/gl/Renderable.java:10]: use `List<Renderable> renderables = new ArrayList<>();`

Comment (Shuzhengz) [reply src/org/dhsdev/flowerknight/gl/Renderable.java:10]: also why public?

**Comment by Shuzhengz** [comment]
ugh nvm i'll create a pr

File: src/org/dhsdev/flowerknight/gl/Renderable.java (hunk 4 above)
Comment (JakeRoggenbuck) [reply src/org/dhsdev/flowerknight/gl/Renderable.java:10]: We can make a pr to fix that

Comment (Camerooooon) [reply src/org/dhsdev/flowerknight/gl/Renderable.java:10]: > also why public?

Because it is accessed in the main class.

**Pull request merged**